### PR TITLE
Fix for iptables states fail when state_aggregate option is enabled

### PR DESCRIPTION
**Pull request by xeacott**
### What does this PR do?
Add method to remove circular references to objects that must be decoded/encoded.

### What issues does this PR fix or reference?
Fix https://github.com/saltstack/salt/issues/53353 by cleaning data object in case of circular references.

### Tests written?
Yes

### Commits signed with GPG?
No

### CODE REVIEW

**Comment by xeacott** [comment]
Forgot to squash when I merged :-1:

**Comment by waynew** [comment]
I'm not entirely sure that I like just removing circular refs. I mean, it probably solves the problem, but it obscures the fact that we shouldn't be creating circular data structures in the first place. At least, I'm not aware of any place in salt where that would be a good thing to have 🤷‍♂

**Comment by xeacott** [comment]
I agree. This just puts in a fail safe that if we do have a circular reference, clean it up and let it pass through. However, I'm not entirely sure where an object would have gained a circular reference, which would be the real root cause. I can investigate it further for sure!

**Comment by waynew** [comment]
I've noticed that we have occasions where this is the case. I wouldn't be opposed to putting this PR in *if* we also logged (probably as an error, probably with the stack trace?) that a circular reference was detected, asking users to report it.

At least, I don't think I would 😂

**Comment by xeacott** [comment]
Let me find the correct place to log and I'll get that in with this PR!

**Review by twangboy — APPROVED** [review]

File: tests/unit/utils/test_data.py [hunk 1]
```diff
@@ -268,6 +268,25 @@ def test_compare_lists_changes_old(self):
...
+        test_dict = {"key": "value", "type": "test1"}
+        test_dict["self"] = test_dict
+        ret = salt.utils.data._remove_circular_refs(ob=test_dict)
+        self.assertDictEqual(ret, {'key': 'value', 'type': 'test1', 'self': None})
+
+    def test_circular_refs_lists(self):
+        test_list = {'foo': [],}
+        test_list['foo'].append((test_list,))
+        ret = salt.utils.data._remove_circular_refs(ob=test_list)
+        self.assertDictEqual(ret, {'foo': [(None,)]})
```
Comment (twangboy) [inline tests/unit/utils/test_data.py]: Looks like this is failing lint. You need a space after the comma.

**Review by twangboy — CHANGES_REQUESTED** [review]

**Pull request closed without merging**